### PR TITLE
Fixes #158959

### LINKED ISSUE #158959: Test merge editor: compare with base should maybe open at the line where the merge editor was
Testing #158819

Not sure if this is expected behavior, but comparing with base opens at the top of the file regardless of where in the merge editor I was. I think it'd be more helpful to open to where I was in the merge editor, since I'd maybe be more interested in what that particular part of the base had?
Recording 2022-08-23 at 11 08 18

### PR DESCRIPTION
**Pull request by hediet**

### CODE REVIEW

File: src/vs/workbench/contrib/mergeEditor/browser/commands/commands.ts [hunk 1]
```diff
@@ -361,20 +362,33 @@ export class CompareInput2WithBaseCommand extends MergeEditorAction {
...
+	const lineNumber = input.getPosition()!.lineNumber;
+	const editor = await editorService.openEditor({
+		original: { resource: base.uri },
+		modified: { resource: input.getModel()!.uri },
+	});
+	const e = editor?.getControl();
+	if (!e) {
+		return;
+	}
+	const d = ((e as any).onDidUpdateDiff as Event<unknown>)(() => {
```
Comment (jrieken) [inline src/vs/workbench/contrib/mergeEditor/browser/commands/commands.ts:384]: This shouldn't be so complicated. I am certain that the `openEditor` call or the input itself supports the passing and revealing of a position. @lramos15 for details

Comment (lramos15) [reply src/vs/workbench/contrib/mergeEditor/browser/commands/commands.ts]: @hediet And I discussed offline and have settled on using the selection options for this

**Review by jrieken — DISMISSED** [review]

File: src/vs/workbench/services/editor/common/editorGroupFinder.ts [hunk 2]
```diff
@@ -205,6 +206,9 @@ function matchesEditor(typedEditor: EditorInput, editor: EditorInput | IUntypedE
 	// editors that have no `override` defined.
 	//
 	// TODO@lramos15 https://github.com/microsoft/vscode/issues/131619
+	if (typedEditor instanceof DiffEditorInput && isResourceDiffEditorInput(editor)) {
```
Comment (lramos15) [inline src/vs/workbench/services/editor/common/editorGroupFinder.ts:209]: @bpasero FYI `revealIfVisible` and `revealIfOpened` were broken with untyped inputs due to doing a top level resource check only which diff editor inputs don't have.

@mjbvz This is yet another reason we hope #131619 can eventually be revisited and addressed. Webviews needing to know the group for the input is a very fragile check on just the resource as we don't know the editor type at this time

**Review by lramos15 — APPROVED** [review]

File: src/vs/workbench/services/editor/common/editorGroupFinder.ts (hunk 2 above)
Comment (bpasero) [reply src/vs/workbench/services/editor/common/editorGroupFinder.ts:209]: Is it possible to have a unit test for this?

Btw could this apply to side by side editors as well, which are not diff inputs but behave very similarly?

**Pull request merged**